### DISCUSSION TITLE
Source maps support in node-loader

### DISCUSSION THREAD

**Discussion by monteeya** [mdx-js/mdx#2240] [General]
I tried to use the experimental node-loader and it works great, except the debugging is very tricky because line numbers don't match.

I'm starting node with "--enable-source-maps", but it doesn't seems that node-loader is currently providing source maps. Is there any workaround to get this working?

**Comment by wooorm** [comment]
AFAIK source maps are super experimental, and node loaders don’t have a way to specify that stuff even.
So, don’t think this is going to work quickly, but, feel free to investigate, maybe I can help!

**Reply by monteeya** [reply]
Thank you for the quick reply.

Yes, source maps in node are still experimental, but they generally work well.

I haven't made a deep dive yet, but I think if we can include source maps in comments in `source` string in node-loader/lib/index.js, everything else should be handled automatically.

**Reply by monteeya** [reply]
I have to test this a bit more, but this seems to work:

I added `SourceMapGenerator` to the compile options and appended the source mapping comment to `source` string in node-loader/lib/index.js:

```
source = `${source}\n//# sourceMappingURL=data:application/json;base64,${Buffer.from(JSON.stringify(file.map)).toString('base64')}`
```

**Reply by wooorm** [reply]
Oh that's very cool. If that works, I'm open to adding it, with some docs maybe?!

**Reply by monteeya** [reply]
I made a few quick tests and it worked well. It definitely requires more testing, but considering that this currently doesn't work at all, I guess it's better than nothing :)
